scopes/dependencies/dependency-resolver/policy/env-policy/env-policy.ts

```diff
@@ -47,23 +48,30 @@ export class EnvPolicy extends VariantPolicy {
     super(_policiesEntries);
   }
 
-  static fromConfigObject(configObject): EnvPolicy {
+  static fromConfigObject(
+    configObject,
+    { includeLegacyPeersInSelfPolicy }: VariantPolicyFromConfigObjectOptions = {}
+  ): EnvPolicy {
     validateEnvPolicyConfigObject(configObject);
 
     /**
      * Calculate the policy for the env itself.
      * Always force it for the env itself
      */
-    const selfPeersEntries = entriesFromKey(configObject, 'peers', 'version', 'runtime', 'env-own', true);
-
+    let selfPeersEntries: VariantPolicyEntry[];
+    if (includeLegacyPeersInSelfPolicy && !configObject.peers && configObject.peerDependencies) {
+      selfPeersEntries = handleLegacyPeers(configObject);
+    } else {
+      selfPeersEntries = entriesFromKey(configObject, 'peers', 'version', 'runtime', 'env-own', true);
+    }
     const selfPolicy = VariantPolicy.fromArray(selfPeersEntries);
 
     /**
      * Legacy policy used by the old getDependencies function on the env aspect.
      * when we used to configure dependencies, devDependencies, peerDependencies as objects of dependencyId: version
      * Those were always forced on the components as visible dependencies.
      */
-    const legacyPolicy = VariantPolicy.fromConfigObject(configObject, 'env', false, true);
+    const legacyPolicy = VariantPolicy.fromConfigObject(configObject, { source: 'env', force: true });
```

Comment:
what about setting hidden as false here?
(not sure it's a must, just want to be sure you validate it is correct when omitting it)

Reply:
checked the code. Better to set it. Fixed.